pkgs/by-name/fi/filius/package.nix

```diff
@@ -0,0 +1,117 @@
+  #   (e.g. https://www.lernsoftware-filius.de/downloads/Setup/filius-2.5.1.zip)
+  # - identify the new version string.
+  version = "2.5.1";
+
+  # - Open http://web.archive.org and paste download link from clipboard into "Save Page Now" field and hit the "Save Page" button.
+  # - Unselect "Save Error Pages" and hit "Save Page" again.
+  # - Wait for the archive link to be generated and copy it to the url field - adjust hash accordingly.
+  src = fetchzip {
+    url = "https://web.archive.org/web/20240623122638/https://www.lernsoftware-filius.de/downloads/Setup/filius-2.5.1.zip";
+    hash = "sha256-UrtmL0jHf86vXXCtRJoaXAyElBvUUF1jGpBcO0GV6Lg=";
+    stripRoot = false;
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
+    copyDesktopItems
+  ];
+
+  desktopItems = [
+    (makeDesktopItem {
+      type = "Application";
+      name = "Filius";
+      desktopName = "Filius";
+      genericName = "Computer Network Simulator";
+      comment = "A Computer Network Simulator for Secondary Schools";
+      icon = "filius";
+      exec = "filius";
+      terminal = false;
+      mimeTypes = [ "application/filius-project" ];
+      categories = [
+        "Education"
+        "Network"
+        "ComputerScience"
+      ];
+      startupNotify = false;
+      extraConfig = {
+        "GenericName[en]" = "Network simulator";
+        "GenericName[de]" = "Netzwerksimulator";
+        "Comment[en]" = "A Computer Network Simulator for Secondary Schools";
+        "Comment[de]" = "Ein Netzwerksimulator für Bildungszwecke";
+      };
+    })
+  ];
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  installPhase = ''
+    runHook preInstall
+
+    export CUSTOM_LIBS=$out/share/java
+    export JAR=$CUSTOM_LIBS/filius.jar
+
+    # "install -D" creates missing folders
+    install -D filius.jar $JAR
+    cp -r . $CUSTOM_LIBS/
+
+    makeWrapper ${jre}/bin/java $out/bin/filius \
+      --add-flags "-Duser.dir=$CUSTOM_LIBS/" \
+      --add-flags "-Xmx512M" \
+      --add-flags "-jar $JAR" \
+      --add-flags '-n -wd $HOME' \
+      --set _JAVA_OPTIONS '-Dawt.useSystemAAFontSettings=lcd'
+
+    runHook postInstall
+  '';
+
+  postInstall = ''
+    install -Dm444 ${icon128} $out/share/icons/hicolor/128x128/apps/filius.png
+  '';
+
+  meta = {
+    homepage = "https://www.lernsoftware-filius.de";
+    downloadPage = "https://www.lernsoftware-filius.de/Herunterladen";
+    description = "A Computer Network Simulator for Secondary Schools";
+    longDescription = ''
+      With the software tool Filius, you can design computer networks yourself,
+      simulate the exchange of messages in them and thus explore their structure
+      and functionality experimentally. The target group are learners at secondary
+      level in general education schools. Filius enables learning activities that
+      are designed to support discovery-based learning in particular.";
+    '';
+    license = with lib.licenses; [
+      gpl2
+      gpl3
```

Comment:
> Filius is provided under the terms of GPLv2 or GPLv3.
> -- <https://gitlab.com/filius1/filius/-/blob/master/Readme.md?ref_type=heads>

not sure but it feels like we should use gpl2Only and gpl3Only. The gpl license specifies that any later gpl version can also be used unless specified otherwise afaik but in this case it is specified